models/forms.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * name : forms.js.
+ * author : Prajwal.
+ * created-date : 08-May-2024.
+ * Description : Schema for Forms.
+ */
+
+const { Schema } = require('mongoose')
+
+const formSchema = new Schema({
+	type: {
+		type: String,
+		required: true,
+		unique: true,
+		index: true,
+	},
+	subType: {
+		type: String,
+		required: true,
+	},
+	version: {
+		type: Number,
+		default: 0,
+	},
+	data: {
+		type: Schema.Types.Mixed,
+	},
+	organizationId: {
+		type: Number,
+		required: true,
+	},
+})
+
+// Pre-update hook to increment version
+formSchema.pre('findOneAndUpdate', function (next) {
```

Comment:
@Prajwal17Tunerlabs this will increment the version only when 'findOneAndUpdate' is called for update right? can we generalise this such that when ever an update happens to a form version will increment? regardless which method we are using. check if it is achievable , if not ignore

Reply:
i did not find any function which would perform our desired operation, made some intuitive/bruteforce changes